src/components/form/examples/form-span-fields.tsx

```diff
@@ -0,0 +1,218 @@
+ * wide field, by two 1 column wide fields, or only partially filled by a single 1 column
+ * wide field. If none of the remaining fields fit, the hole will be left as it is.
+ *
+ * However, you can disable this functionality by setting `dense` to `false` in the
+ * options for the grid layout.
+ *
+ * ```ts
+ * export const schema = {
+ *     type: 'object',
+ *     lime: {
+ *         layout: {
+ *             type: 'grid',
+ *             dense: false,
+ *         },
+ *     },
+ * };
+ * ```
+ *
+ * :::note
+ * Sometimes, the order of fields are important for the way users perceive the form.
+ * If you choose to use the default auto-reordering behavior, make sure to test your
+ * form's layout in different screen sizes to see whether you can mitigate unwanted
+ * layout changes.
+ *
+ * Some unwanted results can be avoided by changing the order of the fields,
+ * so that they render appropriately on different screens, or by dividing
+ * the form into more sections.
+ * :::
+ *
+ * ###### Stretching a field vertically
+ * Most standard elements that can be used in forms, such as `limel-input`,
+ * `limel-select`, `limel-slider`, etc, have a fixed height, and therefore
+ * it does not really make sense to stretch them vertically, and we strongly
+ * recommend you not to!
+ *
+ * But there are some exceptions. One of them is `limel-input-field` with `type='textarea'`.
+ * It's always recommended to assign at least two rows for a textarea, to improve
+ * its usability. Since this is a component that is supposed to show multi-lined
+ * pieces of text.
+ *
+ * Also, if you make create a custom component for your form—let's say
+ * a map—you will have the possibility of using `rowSpan` to increase the height of
+ * your custom component.
+ * ```ts
+ * export const schema = {
+ *     type: 'object',
+ *     properties: {
+ *         comment: {
+ *             type: 'string',
+ *             title: 'Comment',
+ *             lime: {
+ *                 component: {
+ *                     props: {
+ *                         type: 'textarea',
+ *                     },
+ *                 },
+ *                 layout: {
+ *                       colSpan: 3,
+ *                       rowSpan: 2,
+ *                 },
+ *             },
+ *         },
+ *     },
+ * };
+ * ```
+ *
+ * :::warning
+ * Custom web-components that you include in the forms should not have hard-coded
+ * `width` or `height` values! Otherwise they will stretch out of their cell and break
+ * the UI. Make sure that such components are internally designed to be responsive,
+ * and that their `:host` and any potential wrapping container has the following styles:
+ * ```scss
+ * :host {
+ *     display: block; // or another suitable property
+ *     width: 100%;
+ *     height: 100%;
+ * }
+ * :host([hidden]) {
+ *     display: none;
+ * }
+ * .my-component {
+ *     width: 100%;
+ *     height: 100%;
+ * }
+ * ```
+ * :::
```

Comment:
I think the documentation becomes waaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaay too long to navigate easily with all this text and these super long examples… 😞

Maybe we could put them in collapsibles or dialogs or something.

It wasn't a problem when I was reading the whole thing from top to bottom, but now that I was trying to actually find my way around to specific examples to look for one that had a picker or something, it was extremely frustrating.

Reply:
I think that's the downside, or better said the "trade-off" of having guidelines. I always think things are better to be explained well, rather than abstractly mentioned. Specially because we want this to be a stand-alone source of truth for consultants, who have different skill levels.

People can always choose not to read if they want to. But when they actually want to read, there should be content that clarifies things well.

The purpose of "guidelines" is this very thing; not just displaying examples I'd say. So we're satisfying the purpose. If it gets too long, it's not necessarily a problem. (Of course it's better if we can say the same things with less words).

However, when it comes to these forms examples, it feels they can be too long and too many. I don't know if we can merge them somehow. 

Also it'd be nice if Kompendium added more anchor-links to different sections of the documentation and made them available as floating navigation menu somehow, or added them in the nav menu.

Reply:
I mostly agree actually 😄

But, we're mixing api-documentation, examples, and guidelines. It's getting hard to find your way around, and that makes it easy to miss important things.

Adding a way of navigating would improve things, for sure 👍

Also, if we want these longer descriptions in more places, we might want to move the api-docs up, above the examples again, where they were when we used **docz**…

Anyway, it's a work in progress. Sorry for letting my frustration shine through in my comment yesterday. I was tired and just felt "this is completely impossible to work with", but that's not true 😄

Reply:
Great! No worries dude. You absolutely have a point! Maybe we need two separate places for guidelines and actual API stuff. It's probably hard to do sometimes, as these are guidelines about options that consultants have when using the components (which overlap with our "design guidelines" sometimes).